pkg/goanalysis/runner_checker.go

```diff
@@ -90,12 +90,12 @@ func (act *action) analyze() {
 	// Report an error if any dependency failures.
 	var depErrors error
 	for _, dep := range act.Deps {
-		if dep.Err != nil {
+		if dep.Err != nil && errors.Unwrap(dep.Err) != nil {
 			depErrors = errors.Join(depErrors, errors.Unwrap(dep.Err))
```

Comment:
dep.Err could be unwrapped only once

Reply:
nope

Reply:
// Unwrap only calls a method of the form "Unwrap() error".
// In particular Unwrap does not unwrap errors returned by [Join].
func Unwrap(err error) error {
	u, ok := err.(interface {
		Unwrap() error
	})
	if !ok {
		return nil
	}
	return u.Unwrap()
}

if dep.Err is return by Join, it is always nil 

because

func (e *joinError) Unwrap() []error {
	return e.errs
}

Reply:
my point was to somehow raise the fact unwrapping as a cost, and I felt like this could be better

```suggestion
		if dep.Err != nil {
			unwrappedErr := errors.Unwrap(dep.Err)
			if unwrappedErr != nil {
				depErrors = errors.Join(depErrors, unwrappedErr)
			}
```

But you might consider this as a micro optimization

Reply:
Since we only add the unwrapped version, we don't even need the extra check. It's safe to pass `nil` to `errors.Unwrap`.

```go
if unwrappedErr := errors.Unwrap(dep.Err); err != nil {
    depErrors = errors.Join(depErrors, unwrappedErr)
}
```

Reply:
The if `errors.Unwrap` is useless because `errors.Join` ignore `nil`, I will remove it.